File: app/src/main/java/com/innov8/memeit/Fragments/AuthFragment.java
package com.innov8.memeit.Fragments;


import com.innov8.memegenerator.loading_button_lib.customViews.CircularProgressButton;
import com.innov8.memeit.Activities.AuthActivity;

import androidx.fragment.app.Fragment;

public class AuthFragment extends Fragment {

    public AuthFragment() {
    }

    protected CircularProgressButton actionButton;

    protected AuthActivity getAuthActivity(){
        return (AuthActivity) getActivity();
    }

    protected void setLoading(boolean loading){
        if (loading){
            actionButton.startAnimation();
        }
        else {
            actionButton.revertAnimation();

        }
    }
}

File: app/src/main/java/com/innov8/memeit/Fragments/AuthFragment.kt
package com.innov8.memeit.Fragments


import com.innov8.memegenerator.loading_button_lib.customViews.CircularProgressButton
import com.innov8.memeit.Activities.AuthActivity

import androidx.fragment.app.Fragment

open class AuthFragment : Fragment() {
    val onError={message:String->
        setLoading(false)
        authActivity.showError(message)
    }
    protected lateinit var actionButton: CircularProgressButton

    protected val authActivity: AuthActivity
        get() = activity as AuthActivity
    protected var mLoading: Boolean = false
    protected fun setLoading(loading: Boolean) {
        mLoading = loading
        if (loading) {
            actionButton!!.startAnimation()
        } else {
            actionButton!!.revertAnimation()

        }
    }
}
